.github/scripts/evaluate_review_gate.py

```
@@ -14,10 +23,108 @@
 ARCH_VERDICT_RE = re.compile(r"(APPROVE|NEEDS[_ ]REVISION|REJECT)")
-PIPELINE_VERDICT_RE = re.compile(r"^`?PIPELINE_VERDICT:\s*(APPROVE|NEEDS_REVISION|REJECT|CLEAN|NEEDS_REFACTORING|MAJOR_CLEANUP|PASS|FAIL)\b", re.MULTILINE)
+PIPELINE_ARCH_VERDICT_RE = re.compile(r"^`?PIPELINE_ARCH_VERDICT:\s*(APPROVE|NEEDS_REVISION|REJECT)\b", re.MULTILINE)
+PIPELINE_CODE_VERDICT_RE = re.compile(r"^`?PIPELINE_CODE_VERDICT:\s*(CLEAN|NEEDS_REFACTORING|MAJOR_CLEANUP)\b", re.MULTILINE)
 EXIT_STATUS_RE = re.compile(r"^EXIT_STATUS:\s*(SUCCESS|REVISION_LIMIT|USER_INPUT_REQUIRED|ERROR|SKILL_NOT_FOUND)", re.MULTILINE)
 
 
+# ---------------------------------------------------------------------------
+# JSON-based scoring (primary path)
+# ---------------------------------------------------------------------------
+
+def try_json_scoring(json_path: Path) -> dict | None:
```

Ruff (0.15.20)

[warning] Too many return statements (8 > 6)

(PLR0911)